modules/sirdataRtdProvider.js

```diff
@@ -0,0 +1,402 @@
+    gdprApplies = null;
+    tcString = '';
+  } else if (getGlobal().getConfig('consentManagement.gdpr')) {
+  // Default endpoint is cookieless if gdpr management is set. Needed because the cookie-based endpoint will fail and return error if user is located in Europe and no consent has been given
+    sirdataDomain = 'cookieless-data.com';
+    sendWithCredentials = false;
+  }
+
+  // default global endpoint is cookie-based if no rules falls into cookieless or consent has been given or GDPR doesn't apply
+  if (!sirdataDomain || !gdprApplies || (utils.deepAccess(userConsent, 'gdpr.vendorData.vendor.consents') && userConsent.gdpr.vendorData.vendor.consents[53] && userConsent.gdpr.vendorData.purpose.consents[1] && userConsent.gdpr.vendorData.purpose.consents[4])) {
+    sirdataDomain = 'sddan.com';
+    sendWithCredentials = true;
+  }
+
+  var actualUrl = moduleConfig.params.actualUrl || getRefererInfo().referer;
+
+  const url = 'https://kvt.' + sirdataDomain + '/api/v1/public/p/' + moduleConfig.params.partnerId + '/d/' + moduleConfig.params.key + '/s?callback=&gdpr=' + gdprApplies + '&gdpr_consent=' + tcString + (actualUrl ? '&url=' + actualUrl : '');
+  ajax(url, {
+    success: function (response, req) {
+      if (req.status === 200) {
+        try {
+          const data = JSON.parse(response);
+          if (data && data.segments) {
+            addSegmentData(adUnits, data, moduleConfig, onDone);
+          } else {
+            onDone();
+          }
+        } catch (e) {
+          onDone();
+          utils.logError('unable to parse Sirdata data' + e);
+        }
+      } else if (req.status === 204) {
+        onDone();
+      }
+    },
+    error: function () {
+      onDone();
+      utils.logError('unable to get Sirdata data');
+    }
+  },
+  null,
+  {
+    contentType: 'text/plain',
+    method: 'GET',
+    withCredentials: sendWithCredentials,
+    referrerPolicy: 'unsafe-url',
+    crossOrigin: true
+  });
+}
+
+export function setGlobalOrtb2(segments, categories) {
+  try {
+    let addOrtb2 = {};
+    let testGlobal = getGlobal().getConfig('ortb2');
+    if (!testGlobal || !utils.deepAccess(testGlobal, 'user.ext.data.sd_rtd') || !utils.deepEqual(testGlobal.user.ext.data.sd_rtd, segments)) {
+      utils.deepSetValue(addOrtb2, 'user.ext.data.sd_rtd', segments || {});
+    }
+    if (!testGlobal || !utils.deepAccess(testGlobal, 'site.ext.data.sd_rtd') || !utils.deepEqual(testGlobal.site.ext.data.sd_rtd, categories)) {
+      utils.deepSetValue(addOrtb2, 'site.ext.data.sd_rtd', categories || {});
+    }
+    if (!utils.isEmpty(addOrtb2)) {
+      let ortb2 = {ortb2: utils.mergeDeep({}, ((testGlobal && testGlobal.ortb2) ? testGlobal.ortb2 : {}), addOrtb2)};
+      getGlobal().setConfig(ortb2);
+    }
+  } catch (e) {
+    utils.logError(e)
+  }
+
+  return true;
+}
+
+export function setBidderOrtb2(bidder, segments, categories) {
+  try {
+    let addOrtb2 = {};
+    let testBidder = config.getBidderConfig(bidder);
```

Comment:
@nouchy No need to pass the bidder in as this function takes no arguments. I would think you should be able to do something similar to what I commented on the global fpd to reduce the number of validations and the need to look for ortb2.

`let testBidder = utils.deepAccess(config.getBidderConfig(), `${bidder}.ortb2`) || {};`

So for example, line 113 should be able to be reduced to:

`if (!utils.deepAccess(testBidder, 'user.ext.data.sd_rtd') || !utils.deepEqual(testBidder.user.ext.data.sd_rtd, segments)) {...`

Then 116 can be reduced similarly for site data and 120 would simply need testBidder as the second argument of mergeDeep as opposed to the validations

Reply:
Done :) Thank you !